src/legacy/views/secondary-bar/parts/edit/share-folder-properties.tsx

```diff
@@ -145,35 +162,33 @@ const Grantee = ({
 				onMouseLeave={onMouseLeave}
 				onMouseEnter={onMouseEnter}
 				grant={grant}
-				createSnackbar={createSnackbar}
-				folders={folders}
-				allCalendars={allCalendars}
-				setModal={setModal}
 				setActiveModal={setActiveModal}
-				totalAppointments={totalAppointments}
-				setActiveGrant={setActiveGrant}
 			/>
 		</Container>
 	);
 };
 
 export const ShareFolderProperties = ({
 	folder,
-	createSnackbar,
-	folders,
-	allCalendars,
-	setModal,
-	setActiveGrant,
-	totalAppointments,
 	setActiveModal
-}) => {
+}: ShareFolderPropertiesProps): React.JSX.Element => {
 	const [t] = useTranslation();
-	const { grant } = folder.sharedWith;
+	const [grant, setGrant] = useState<Array<Grant> | undefined>();
 	const shareFolderRoleOptions = useMemo(
-		() => ShareFolderRoleOptions(t, grant.perm?.includes('p')),
-		[t, grant.perm]
+		() => ShareFolderRoleOptions(t, grant?.[0]?.perm?.includes('p')),
+		[t, grant]
 	);
-
+	useEffect(() => {
+		soapFetch('GetFolder', {
+			_jsns: 'urn:zimbraMail',
+			folder: { l: folder.id }
+		}).then((res): void => {
+			const response = res as GetFolderActionResponse;
```

Comment:
This cast should not be required, `soapFetch` allow setting generics to get a typed response